<!--

@license Apache-2.0

Copyright (c) 2018 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# no-self-require

> [ESLint rule][eslint-rules] enforcing that modules do not require themselves.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
const rule = require( '@stdlib/_tools/eslint/rules/no-self-require' );
```

#### rule

[ESLint rule][eslint-rules] enforcing that modules do not require themselves.

**Bad**:

<!-- run-disable -->

<!-- eslint-disable stdlib/no-self-require, stdlib/no-dynamic-require -->

```javascript
const myself = require( __filename );
```

**Good**:

<!-- run-disable -->

<!-- eslint-disable stdlib/require-file-extensions -->

```javascript
const other = require( './other.js' );
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint-disable stdlib/require-file-extensions -->

<!-- eslint no-undef: "error" -->

```javascript
const resolve = require( 'path' ).resolve;
const Linter = require( 'eslint' ).Linter;
const rule = require( '@stdlib/_tools/eslint/rules/no-self-require' );

const linter = new Linter();

const code = 'var myself = require( \'./index.js\' );';

linter.defineRule( 'no-self-require', rule );

const result = linter.verify( code, {
    'rules': {
        'no-self-require': 'error'
    }
}, {
    'filename': resolve( './index.js' )
});
/* returns
    [
        {
            'ruleId': 'no-self-require',
            'severity': 2,
            'message': 'module requires itself',
            'line': 1,
            'column': 14,
            'nodeType': 'CallExpression',
            'source': 'var myself = require( \'./index.js\' );',
            'endLine': 1,
            'endColumn': 37
        }
    ]
*/
```

</section>

<!-- /.examples -->

<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->

<section class="related">

</section>

<!-- /.related -->

<!-- Section for all links. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
