<!--

@license Apache-2.0

Copyright (c) 2024 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# require-spaces

> [ESLint rule][eslint-rules] enforcing spaces in `require()` statements.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
const rule = require( '@stdlib/_tools/eslint/rules/require-spaces' );
```

#### rule

[ESLint rule][eslint-rules] enforcing spaces in `require()` statements.

**Good**:

```javascript
const betainc = require( '@stdlib/math/base/special/betainc' );
```

**Bad**:

<!-- eslint-disable stdlib/require-spaces -->

```javascript
const betainc = require('@stdlib/math/base/special/betainc');
const zip = require( '@stdlib/utils/zip');
const noop = require('@stdlib/utils/noop' );
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
const Linter = require( 'eslint' ).Linter;
const rule = require( '@stdlib/_tools/eslint/rules/require-spaces' );

const linter = new Linter();

const code = 'var betainc = require(\'@stdlib/math/base/special/betainc\');';

linter.defineRule( 'require-spaces', rule );

const result = linter.verify( code, {
    'rules': {
        'require-spaces': 'error'
    }
});
console.log( result );
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
